src/test/java/net/kaczmarzyk/IsEmptyE2eTest.java

```diff
@@ -0,0 +1,106 @@
+import net.kaczmarzyk.spring.data.jpa.domain.IsEmpty;
+import net.kaczmarzyk.spring.data.jpa.web.annotation.Spec;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import static net.kaczmarzyk.spring.data.jpa.CustomerBuilder.customer;
+import static net.kaczmarzyk.spring.data.jpa.OrderBuilder.order;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+/**
+ * @author Hubert Gotfryd (Tratif sp. z o.o.)
+ */
+public class IsEmptyE2eTest extends E2eTestBase {
+
+    @Controller
+    public static class IsEmptySpecController {
+
+        @Autowired
+        CustomerRepository customerRepo;
+
+        @RequestMapping(value = "/customersWithEmptyOrders")
+        @ResponseBody
+        public Object findByEmptyOrders(
+                @Spec(path="orders", spec=IsEmpty.class) Specification<Customer> spec) {
+
+            return customerRepo.findAll(spec);
+        }
+
+        @RequestMapping(value = "/customersWithEmptyPhoneNumbers")
+        @ResponseBody
+        public Object findByEmptyPhoneNumbers(
+                @Spec(path="phoneNumbers", spec=IsEmpty.class) Specification<Customer> spec) {
+
+            return customerRepo.findAll(spec);
+        }
+    }
+
+    @Before
+    public void initData() {
+        customer("Barry", "Benson")
+                .phoneNumbers("123456789")
+                .build(em);
+        customer("Vanessa", "Bloom")
+                .orders(order("flowers"))
+                .build(em);
+    }
+
+    @Test
+    public void findsByEmptyCustomerOrders_oneToManyAssociation() throws Exception {
+        mockMvc.perform(get("/customersWithEmptyOrders")
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$").isArray())
+                .andExpect(jsonPath("$[0].firstName").value("Marge"))
+                .andExpect(jsonPath("$[1].firstName").value("Lisa"))
+                .andExpect(jsonPath("$[2].firstName").value("Maggie"))
+                .andExpect(jsonPath("$[3].firstName").value("Minnie"))
+                .andExpect(jsonPath("$[4].firstName").value("Barry"))
+                .andExpect(jsonPath("$[5]").doesNotExist());
+    }
+
+    @Test
+    public void findsByEmptyCustomerPhoneNumbers_elementCollectionWithSimpleNonEntityValues() throws Exception {
+        mockMvc.perform(get("/customersWithEmptyPhoneNumbers")
```

Comment:
Is in DB any customers with non empty phone numbers?

Reply:
`Barry Benson` saved before test